### DISCUSSION TITLE
Interface between mainboard and extruder board

### DISCUSSION THREAD

**Discussion by Blitzboot** [xblax/flashforge_ad5m_klipper_mod#212] [Development]
Hello everyone, 
I would like to build my own printer and have bought the Flashforge ad5m extruder for this purpose. I would like to use a Raspberry and an Arduino to control it. I don't have the mainboard of the printer. However, the extruder itself also has a board and I am now faced with the problem that I don't know how to control the extruder. The pins that run into the extruder board are 2x ground 2x 24V and TX and RX. 
Does anyone know the standard or the interface that is used for communication here? 

I hope someone can help me and I have explained the problem clearly. 

Mark

**Comment by KaruroChori** [comment]
CANBUS. You will need a serial to CANBUS adapter on your raspberry. They are extremely cheap.
However this discussion is not really on topic with this repository.

**Reply by Blitzboot** [reply]
Thank you for your Answer. And i Know but i searched the Web and this was the closest..

**Reply by KaruroChori** [reply]
Be aware that this toolhead comes with an old klipper version installed. And upgrading it is not really viable. So you will be limited to work with klipper 0.11 at best.
I would not suggest it for custom builds.

**Reply by Blitzboot** [reply]
I just need to send commands to the Extruder for temp. und Extrusion (Stepper). Do you think i can use it for this? otherwise I can remove the Board of the Extruder and get a new driver for the stepper motor and control it directly with the Arduino and the raspberry

**Reply by akarkou1** [reply]
the toolhead does not use canbus. it uses a rs232 transceiver (3PEAK TPT2323E).
